web/src/date-utils.ts

```diff
@@ -66,7 +67,14 @@ export const dateToFormat = (date: Date | number, format: DateFormat): string =>
       case DateFormat.TimeFull: {
         const fractionalSeconds =
           typeof date === 'number' ? Math.floor(date % 1000) : date.getMilliseconds();
-        return `${timeMedFormatter.format(date)}.${fractionalSeconds}`;
+        const parts = timeMedFormatter.formatToParts(date);
+        const hours = parseInt(getPart(parts, 'hour'), 10);
+        const clampedHours = !isNaN(hours) ? padLeadingZero(hours >= 24 ? 0 : hours) : '';
+
+        return `${clampedHours}:${getPart(parts, 'minute')}:${getPart(
+          parts,
+          'second',
+        )}.${fractionalSeconds}`;
```

Comment:
Maybe it's easier to just not use `Intl.DateTimeFormat`'s `formatToParts` in this case?

Something like (not tested)

```
const d = new Date(date);
const pad = (number: number) => (number < 10 ? `0${number}` : number);
return `${pad(d.getHours())}:${pad(d.getMinutes())}:${pad(d.getSeconds())}.${fractionalSeconds}`;
```

Reply:
yes I agree , I refactored to use `Intl.DateTimeFormat` on the only case is needed.